### PR TITLE
[improvement] improve URL validation in branding and move HTML text sanitization to `core`

### PR DESCRIPTION
**Pull request by Achintha444**
### Purpose
[improvement] improve URL validation in branding and move HTML text sanitization to `core`

### Related PRs
- https://github.com/wso2/identity-apps/pull/5599
- https://github.com/wso2/identity-apps/pull/5592

### Checklist
- [ ] e2e cypress tests locally verified.
- [ ] Manual test round performed and verified.
- [ ] UX/UI review done on the final implementation.
- [ ] Documentation provided. (Add links if there are any)
- [ ] Unit tests provided. (Add links if there are any)
- [ ] Integration tests provided. (Add links if there are any)

### Security checks
- [ ] Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines?
- [ ] Ran FindSecurityBugs plugin and verified report?
- [ ] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets?

### CODE REVIEW

**Review by pavinduLakshan — COMMENTED** [review]
Is it intentional to not commit the lockfile?

**Comment by Achintha444** [comment]
> Is it intentional to not commit the lockfile?

Committed the lock file

File: apps/console/src/features/branding/components/advanced/advance-form.tsx [hunk 1]
```diff
@@ -155,11 +156,7 @@ export const AdvanceForm: FunctionComponent<AdvanceFormPropsInterface> = forward
         // Use a regex to replace {{lang}}, {{country}}, and {{locale}} placeholders while preserving other characters
         moderatedValue = value?.trim().replace(new RegExp(placeholdersPattern, "g"), "");
 
-        // Gracefully handle scripting attempts.
-        // TODO: This should be moved to a common util or validation library.
-        const javascriptPattern: string = "javascript:";
-
-        if (moderatedValue?.includes(javascriptPattern) || !FormValidation.url(moderatedValue)) {
+        if (URLUtils.isURLValid(moderatedValue) || !FormValidation.url(moderatedValue)) {
```
Comment (pavinduLakshan) [inline apps/console/src/features/branding/components/advanced/advance-form.tsx:159]: Isn't the following the correct one?

```suggestion
        if (!URLUtils.isURLValid(moderatedValue) || !FormValidation.url(moderatedValue)) {
```

File: modules/core/src/utils/index.ts [hunk 2]
```diff
@@ -21,6 +21,7 @@ export * from "./certificate-management-utils";
 export * from "./common-utils";
 export * from "./context-utils";
 export * from "./crypto-utils";
+export * from "./encode";
 export * from "./encode-decode-utils";
```
Comment (pavinduLakshan) [inline modules/core/src/utils/index.ts:25]: Wouldn't having two files with `encode` term cause confusion? Also, the term `encoding` doesn't correctly convince what we are doing here.

IMO, it'd better to use a term such as `sanitize`, `escape` etc.

File: modules/core/src/utils/url-utils.ts [hunk 3]
```diff
@@ -108,31 +107,38 @@ export class URLUtils {
...
+                protocol // https|http
             } as URLComponentsInterface;
         } catch (error) {
             return null;
         }
     }
 
     public static isURLValid(url: string, checkForSanity?: boolean): boolean {
+        // List of invalid keywords that cannot be present in a URL.
+        const invalidKeywords: string[] = [ "javascript:" ];
```
Comment (pavinduLakshan) [inline modules/core/src/utils/url-utils.ts:134]: sanitizeUrl method already handles this, doesn't it?

File: modules/core/src/utils/index.ts (hunk 2 above)
Comment (Achintha444) [reply modules/core/src/utils/index.ts:25]: cc @brionmario WDYT ?

File: modules/core/src/utils/url-utils.ts (hunk 3 above)
Comment (Achintha444) [reply modules/core/src/utils/url-utils.ts:134]: It does not handle these cases

File: apps/console/src/features/branding/components/advanced/advance-form.tsx (hunk 1 above)
Comment (Achintha444) [reply apps/console/src/features/branding/components/advanced/advance-form.tsx]: Fixed with https://github.com/wso2/identity-apps/pull/5611/commits/72ff1d82a3026daad4df90f5686292b721900dbd

File: modules/core/src/utils/url-utils.ts (hunk 3 above)
Comment (Achintha444) [reply modules/core/src/utils/url-utils.ts]: Fixed with https://github.com/wso2/identity-apps/pull/5611/commits/c8443438e56f45c40c02e94398280194200ed540

**Pull request merged**